lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -17,72 +17,81 @@
 send log to Loki by implementing the consumer side of the `loki_push_api` relation interface.
 For instance a Promtail or Grafana agent charm that needs to send logs to Loki.
 
+- `LogProxyConsumer`: This object can be used by any K8s charmed operator that needs to
+send logs to Loki through a Log Proxy by implementing the consumer side of the `loki_push_api`
+relation interface.
+Filtering logs in Loki is largely performed on the basis of labels.
+In the Juju ecosystem, Juju topology labels are used to uniquely identify the workload that
+generates telemetry like logs.
+In order to be able to control the labels on the logs pushed this object injects a Pebble layer
+that runs Promtail in the workload container, injecting Juju topology labels into the
+logs on the fly.
 
 
 ## LokiPushApiProvider Library Usage
 
-This object may be used by Loki charms to manage relations
-with their clients.
-For this purposes a Loki charm needs to instantiate the
-`LokiPushApiProvider` object providing it with two pieces
-of information:
+This object may be used by any Charmed Operator that implements the `loki_push_api` interface,
+for instance Loki or Grafana Agent.
+For this purposes a charm needs to instantiate the `LokiPushApiProvider` object with one mandatory
+and three optional arguments.
 
-- A reference to the parent (Loki) charm.
+- `charm`: A reference to the parent (Loki) charm.
 
-- Optionally, the name of the relation that the Loki charm uses to interact
-  with its clients. If provided, this relation name must match a provided
+- `relation_name`: The name of the relation that the charm uses to interact
+  with its clients which implements `LokiPushApiConsumer` or `LogProxyConsumer`.
+  If provided, this relation name must match a provided
   relation in metadata.yaml with the `loki_push_api` interface.
-  This argument is not required if your metadata.yaml has precisely one
-  provided relation in metadata.yaml with the `loki_push_api` interface, as the
-  lib will automatically resolve the relation name inspecting the using the
-  meta information of the charm.
+  Typically `LokiPushApiConsumer` use "logging" as a relation_name and `LogProxyConsumer` use
+  "log_proxy".
```

Comment:
Not sure if this is the right place or if was already mentioned elsewhere: I'd add an explicit differentiation between push api and logproxy, for example:
push api is for talking directly to loki, whereas log proxy is for sending logs via promtail.
(BTW how does syslog fit in here?)

Reply:
@dstathis what do you think about this?

Reply:
Given that this section is for the provider, it should always be logging. You don't provide log_proxy. You only consume it.

Reply:
(syslog is for log_proxy only and is discussed later)